lib/Doctrine/Annotations/Metadata/Builder/PropertyMetadataBuilder.php

```diff
@@ -23,7 +23,7 @@ final class PropertyMetadataBuilder
     /** @var bool */
     private $default = false;
 
-    /** @var array<int|float|string|bool>|null */
+    /** @var array<string, array<int|float|string|bool>|string> $enum */
```

Comment:
Not sure if this is acceptable to you, please tell me what format you prefer (maybe mixed[] + a text description in some format). The actual form of this property is `['value' => [1,2,3], 'literal' => '1,2,3']`

Reply:
property can be null - in case when `withEnum` is not called

Reply:
and cannot be string, because only array is written to it

Reply:
The `|string` is inside the bracket

Reply:
Ah, I see, you specified the inside array :-) ok then